freva-client/src/freva_client/auth.py

```diff
@@ -0,0 +1,195 @@
+"""Module that handles the authentication at the rest service."""
```

Comment:
Ok, I'll give authlib a try. It indeed reduces the code quite a bit. 

Generally speaking. I think we should stick to a serial approach when I comes to PR's. One person submits a PR and changes the code. Others make their review and request changes. If we switch roles back and forth this is going to end up in confusion. So in short, finish one PR rather than other PR's that are related to this PR.

Reply:
Ok, I had a look into authlib. It looks quite interesting for django @Karinon ?

But I don't think that we would benefit from it here for two reasons:

- Looking at the [dependencies](https://github.com/lepture/authlib/blob/11f13e4070c20fffe9c713440d7ce835c686935e/pyproject.toml#L5) I see that it needs `cryptography` which can be quite hard to setup. I'd be hesitant to add it to the client requirements. Because users should be able to install this lib smoothly. I wouldn't care to much about this if we would be talking about the freva-rest or data-loader lib, as this is setup by us. But the client lib is different.
- You give an example?  
   ```python
   OAuth2Session(client_id, client_secret, token_endpoint=f"{cfg.auth_url}/token")
   ```
   How should the users know what client id and client secret to use? All the users should know, is the url of the API, that's it. I don't want to start with a config file where we have to add things so the client lib knows what to do. All of this information should be stored server side. The user should only know where the server is. Ok username and password are obvious. 
   
  We can aim at creating new endpoints that follow the OAuth2Session protocol, which would be beneficial. But I think that's something for another PR.